tests/scanner/audit/instance_network_interface_engine_test.py

```diff
@@ -0,0 +1,92 @@
+# Copyright 2017 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Tests the EnforcedNetworkRulesEngine."""
+
+import copy
+import itertools
+import mock
+import yaml
+
+from google.apputils import basetest
+from google.cloud.security.common.util import file_loader
+from google.cloud.security.scanner.audit.errors import InvalidRulesSchemaError
+from google.cloud.security.scanner.audit import base_rules_engine as bre
+from google.cloud.security.scanner.audit import instance_network_interface_rules_engine as ini
+from google.cloud.security.scanner.audit import rules as scanner_rules
+from tests.unittest_utils import get_datafile_path
+
+
+# TODO: Define more tests
```

Comment:
Please add tests showing violations work correctly for all passing and error cases.

Networks in whitelist and allowed projects.
Network in allowed project but not whitelist, with external IP.
Network in allowed project but not whitelist, with no external IP.
Network not in allowed project.